### ISSUE TITLE
String literals in -eq comparison not caught

### ISSUE THREAD

**Issue by evanjardineskinner** [koalaman/shellcheck#1129]
#### For bugs
- SC2130
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

declare -A WEBHOOK_VALUE
UUID=1
WEBHOOK_VALUE["$UUID"]=1

if [[ "WEBHOOK_VALUE[$UUID]" -eq 1 ]]
then
	echo ohno
fi
```

#### Here's what shellcheck currently says:

Line 5:
WEBHOOK_VALUE["$UUID"]=1
^-- SC2034: WEBHOOK_VALUE appears unused. Verify it or export it.

#### Here's what I wanted or expected to see:

The above but also an SC2130 for the line:

```
if [[ "WEBHOOK_VALUE[$UUID]" -eq 1 ]]
```

as the LHS is a string and cannot be a number, the LHS  should be:

```
if [[ "${WEBHOOK_VALUE[$UUID]}" -eq 1 ]]
```

On older versions of shellcheck this used to be caught. version 0.3.7 catches it

**Comment by koalaman** [comment]
0.3.7 was wrong, this is in fact valid bash:

   UUID=42
   WEBHOOK_VALUES[42]=1
   [[ "WEBHOOK_VALUE[$UUID]" -eq 1 ]] && echo "this prints"

The bug is that ShellCheck doesn't realize that the LHS is a reference to an array.

**Comment by evanjardineskinner** [comment]
But the LHS isn't an array reference it is a string literal with a var reference in it to $UUID. To be an array reference it has to have the leading $ and  the curly braces around WEBHOOK_VALUE. See the bash output below:

```

**Comment by evanjardineskinner** [comment]
Oh I see what you mean, echo prints out as a string literal but the when used inside [[ with -eq it treats it as an array. Awesome thank you. I need to get my build macine updated form 0.3.7 to the latest :-)

**Comment by evanjardineskinner** [comment]
My confidence in Shellcheck has been well and truly reaffirmed. Thank you for all your work on Shellcheck!
